src/qtt/algorithms/functions.py

```diff
@@ -381,33 +404,43 @@ def estimate_dominant_frequency(signal, sample_rate=1, remove_dc=True, fig=None)
     """ Estimate dominant frequency in a signal
 
     Args:
-        signal (array): Input data
+        signal (numpy array): Input data
         sample_rate (float): Sample rate of the data
         remove_dc (bool): If True, then do not estimate the DC component
         fig (int or None): Optionally plot the estimated frequency
     """
     w = np.fft.fft(signal)
     freqs = np.fft.fftfreq(len(signal), d=1. / sample_rate)
+    select_pos = freqs >= 0
+    w_pos = w[select_pos]
+    freqs_pos = freqs[select_pos]
 
     if remove_dc:
-        w[0] = 0
+        if freqs_pos[0] == 0.0:
+            w_pos[0] = 0
+        dc_index = np.argmin(np.abs(freqs))
+        if freqs[dc_index] == 0.0:
+            w[0] = 0
 
     ff = freqs[np.argmax(np.abs(w))]
+    ff_pos = freqs_pos[np.argmax(np.abs(w_pos))]
 
     if fig:
         plt.figure(fig)
         plt.clf()
-        plt.plot(freqs, np.abs(w), '.b')
+        plt.plot(freqs, np.abs(w), '.r')
+        plt.plot(freqs_pos, np.abs(w_pos), '.b')
         plt.xlabel('Frequency')
         plt.ylabel('Abs of fft')
-        plot_vertical_line(ff)
-    return ff
+        plot_vertical_line(ff, color='r')
```

Comment:
```suggestion

```
Not not plot this value. It is not used.

Reply:
Then we do not need to calculate it either (few lines above)